lib/matplotlib/widgets.py

```diff
@@ -529,6 +529,25 @@ def on_changed(self, func):
         """
         return self._observers.connect('changed', lambda val: func(val))
 
+    def set_limits(self, vmin=None, vmax=None):
```

Comment:
I think you may be able to just put this on the `SliderBase` object. Or least share some of what you are doing in the SliderBase superclass.

Reply:
@timhoffm  @ianhi Should I push this functionality to SliderBase ? 
Moreover how can I test that #19265 and this PR don't affect each other ?

Reply:
> @timhoffm @ianhi Should I push this functionality to SliderBase ?

I'd say so. There are several parts that should be shared between all types of sliders. You may need to add some specific parts in the subclass method. So it would end up looking like:

```python
class SliderBase:

     def set_limits(self, vmin, vmax)
...

class Slider:

     def set_limits(self, vmin, vmax):
          super().set_limits(vmin, vmax)
          # other stuff
```